### ISSUE TITLE
Illegal memory access 

### ISSUE THREAD

**Issue by nangchoo** [paclopes/HungarianGPU#3]
In this line (https://github.com/paclopes/HungarianGPU/blob/master/HungarianCUDA.cu#L620), the second argument `&d_min_in_mat` is the pointer to a variable whose type is `data`.

But, In this line (https://github.com/paclopes/HungarianGPU/blob/master/HungarianCUDA.cu#L594), you access it as if it's an array. Wouldn't this be a bug?

**Comment by paclopes** [comment]
No. In the code the blockIdx.x is always zero, so it only acesses the one value.
